### PR TITLE
chore: speedup lsp tests

### PR DESCRIPTION
**Pull request by dos65**
`ClasspathSearch.fromPackages` call with an empty clasppath takes
about a second on my machine.
In theory, replacing it by `empty` should save 4-7 minutes on CI.

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]

File: tests/unit/src/main/scala/tests/TestingServer.scala [hunk 1]
```diff
@@ -1688,4 +1691,37 @@ object TestingServer {
       treeViewProvider = Some(true),
       slowTaskProvider = Some(true)
     )
+
+  // In most cases lsp unit tests have the same classpath: one jvm + scala-library without any external deps
+  // Than means that in all these tests ClasspathSearch should be the same and we can calculate it once per run
+  // instead of once per test
+  val testingClasspathSearchIndexer: ClasspathSearch.Indexer =
```
Comment (dos65) [inline tests/unit/src/main/scala/tests/TestingServer.scala:1698]: Added an additional improvement by reducing the time for ClasspathSearch indexing

File: metals/src/main/scala/scala/meta/internal/metals/Compilers.scala [hunk 2]
```diff
@@ -172,8 +172,10 @@ class Compilers(
   }
 
   def load(paths: Seq[AbsolutePath]): Future[Unit] =
-    if (Testing.isEnabled) Future.successful(())
-    else {
+    if (Testing.isEnabled) {
+      scribe.info(s"LOAD $paths")
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/Compilers.scala:176]: Is this log needed? Could we maybe adjust the message to show what we are printing?

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 3]
```diff
@@ -163,8 +165,9 @@ class MetalsLanguageServer(
   private val recentlyFocusedFiles = new ActiveFiles(time)
   private val languageClient = new DelegatingLanguageClient(NoopLanguageClient)
   var userConfig: UserConfiguration = UserConfiguration()
-  val excludedPackageHandler: ExcludedPackagesHandler =
-    new ExcludedPackagesHandler(userConfig.excludedPackages)
+  var excludedPackageHandler: ExcludedPackagesHandler =
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:168]: Why do we need to have it as a var ? Can't we keep the state inside ExcludedPackagesHandler?

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 4]
```diff
@@ -1049,10 +1053,12 @@ class MetalsLanguageServer(
     // Update in-memory buffer contents from LSP client
     buffers.put(path, params.getTextDocument.getText)
 
-    packageProvider
-      .workspaceEdit(path)
-      .map(new ApplyWorkspaceEditParams(_))
-      .foreach(languageClient.applyEdit)
+    timerProvider.timedThunk("package", true) {
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1056]: Is this needed still? Second parameter is anyway true
```suggestion
    timerProvider.timedThunk("package") {
```

File: tests/unit/src/main/scala/tests/TestingServer.scala (hunk 1 above)
Comment (tgodzik) [reply tests/unit/src/main/scala/tests/TestingServer.scala:1698]: Aren't there any unit tests that have dependencies and the classpath search would be influenced by this?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Great to see these improvements! :tada: 

I added a few questions about the approach.

File: metals/src/main/scala/scala/meta/internal/metals/Compilers.scala (hunk 2 above)
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/Compilers.scala]: oops, just leftovers from investigations. removed

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala (hunk 3 above)
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:168]: We can. I did it in the same manner as other things in MetalsLanguageServer

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala (hunk 4 above)
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: removed

File: tests/unit/src/main/scala/tests/TestingServer.scala (hunk 1 above)
Comment (dos65) [reply tests/unit/src/main/scala/tests/TestingServer.scala:1698]: There are and they work fine with this change.
Cache is done by a key: dependency jars + excludedPackages conf + bucket size.
Unit test with a unique dependencies creates a new index but most of them reuses the default.

Comment (tanishiking) [reply tests/unit/src/main/scala/tests/TestingServer.scala:1698]: Can we add your comment to the code comment?

> There are and they work fine with this change.
> Cache is done by a key: dependency jars + excludedPackages conf + bucket size.
> Unit test with a unique dependencies creates a new index but most of them reuses the default.

**Review by tgodzik — APPROVED** [review]
LGTM!

File: tests/unit/src/main/scala/tests/TestingServer.scala (hunk 1 above)
Comment (dos65) [reply tests/unit/src/main/scala/tests/TestingServer.scala:1699]: updated this comment

**Review by tanishiking — APPROVED** [review]
LGTM :+1:

**Pull request merged**